native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/ArgumentHandler.java

```diff
@@ -161,6 +196,111 @@ private Object getArgumentValue(BObject argumentNode, Type parameterType) {
         }
     }
 
+    private Object getIdArgumentValue(BObject argumentNode, Type parameterType) {
+        if (argumentNode.get(VALUE_FIELD) instanceof BString) {
+            String obj = ((BString) argumentNode.get(VALUE_FIELD)).getValue();
+            if (parameterType.getTag() == TypeTags.STRING_TAG) {
+                return StringUtils.fromString(obj);
+            } else if (parameterType.getTag() == TypeTags.INT_TAG) {
+                return Integer.parseInt(obj);
+            } else if (parameterType.getTag() == TypeTags.FLOAT_TAG) {
+                return ValueUtils.convert(JsonUtils.parse(obj), parameterType);
+            } else if (parameterType.getTag() == TypeTags.DECIMAL_TAG) {
+                return ValueUtils.convert(JsonUtils.parse(obj), parameterType);
+            } else if (parameterType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG) {
+                // not validating if this is uuid:Uuid since compiler plugin does that
+                return ValueCreator.createRecordValue(parameterType.getPackage(), parameterType.getName(),
+                        (BMap<BString, Object>) JsonUtils.parse(obj.replaceAll("\\\\", "")));
+            } else if (parameterType.getTag() == TypeTags.ARRAY_TAG) {
+                return getArrayTypeIdValue(parameterType, obj);
+            } else if (parameterType.getTag() == TypeTags.UNION_TAG) {
+                return getUnionTypeIdValue(parameterType, argumentNode);
+            }
+        }
+        return argumentNode.get(VALUE_FIELD);
+    }
+
+    private Object getUnionTypeIdValue(Type parameterType, BObject argumentNode) {
+        List<Type> members = ((UnionType) parameterType).getMemberTypes();
+        for (Type member : members) {
+            if (member.getTag() != TypeTags.NULL_TAG && member.getTag() != TypeTags.ERROR_TAG) {
+                return getIdArgumentValue(argumentNode, member);
+            }
+        }
+        return null;
+    }
+
+    private Object getArrayTypeIdValue(Type parameterType, String obj) {
```

Comment:
If the parameter type is an array, shouldn't it get an array of IDs?

Reply:
Only string and int are allowed to pass as ID type, if it is an array it has to be passed as a string
eh: int[]
"[2, 3, 4]"

Reply:
<img width="538" alt="Screenshot 2023-06-15 at 13 09 42" src="https://github.com/ballerina-platform/module-ballerina-graphql/assets/39673615/1831276b-220f-4954-925c-0dfe67b33ff2">

Reply:
<img width="1174" alt="Screenshot 2023-06-15 at 13 10 59" src="https://github.com/ballerina-platform/module-ballerina-graphql/assets/39673615/2f7aa0d2-c7a1-46f7-a601-ca22305cf70d">

Reply:
string array
<img width="909" alt="Screenshot 2023-06-15 at 13 21 35" src="https://github.com/ballerina-platform/module-ballerina-graphql/assets/39673615/c8db8aee-946d-4df1-8b7b-a1af7df1d8e5">

Reply:
This seems like an issue in the schema generation. 

For the following resource:

```
resource function get names(@graphql:ID int[] ids) returns string[] {}
```

The generated fields should be this:

```
type Query {
    names(ids: [Int!]!): [String!]!
}
```